lib/team_statistics.rb

```diff
@@ -0,0 +1,310 @@
+  end
+
+  def worst_season(team_id)
+    worst = {}
+    total_loss_by_season(team_id).each do |season_id, loss|
+      worst[season_id] = ((loss.to_f/total_games_by_season(team_id)[season_id]) * 100)
+    end
+    worst.max_by {|season_id, loss_percentage| loss_percentage}.to_a[0]
+  end
+
+# [x] 4. Average Win Percentage
+  def total_games(team_id)
+    #returns for team_id '6' = 7.0
+    season_win_data_set.find_all do |team|
+      team[1] == team_id
+    end.count.to_f
+  end
+
+  def winning_games(team_id)
+    #returns for team_id '6' = 4.0
+    season_win_data_set.find_all do |team|
+      team[1] == team_id && team[2] == "WIN"
+    end.count.to_f
+  end
+
+  def average_win_percentage(team_id)
+    "#{(winning_games(team_id)/total_games(team_id) * 100).round(2)}"
+  end
+
+# [x] 5/6 Most and Fewest Goals Scored
+  def score_data_set
+    @stat_tracker[:game_teams]["team_id"].zip(@stat_tracker[:game_teams]["goals"])
+  end
+
+  def team_per_game(team_id)
+    #returns for team_id '6' [["6", "3"], ["6", "3"], ["6", "1"], ["6", "3"], ["6", "4"], ["6", "2"], ["6", "3"]]
+    score_data_set.find_all do |team|
+      team[0] == team_id
+    end
+  end
+
+  def most_goals_scored(team_id)
+    team_per_game(team_id).max_by do |goal|
+      goal[1]
+    end[1].to_i
+  end
+
+  def fewest_goals_scored(team_id)
+    team_per_game(team_id).min_by do |goal|
+      goal[1]
+    end[1].to_i
+  end
+
+# [ ] 7/8 Favorite and Rival Opponent **change name of 'season_win_data_set'
+  def team_id_to_team_name(team_id)
+    team_name_lookup = stat_tracker[:teams]["team_id"].zip(stat_tracker[:teams]["teamName"]).to_h
+    team_name_lookup[team_id]
+  end
+
+  def favorite_opponent(team_id)
+    #evaluate all teams that given team has played and won
+    papa_bear = {}
+    mama_bear ={}
+
+    games_won = season_win_data_set.find_all do |game_id|
+      team_id == game_id[1] && game_id[2] == "WIN"
+    end
+    winning_game_ids = games_won.map do |set|
+      set.first
+    end
+    opposing_teams = winning_game_ids.map do |game_id|
+      season_win_data_set.find_all do |array|
+        game_id == array[0] unless array[1] == team_id
+      end.flatten
+    end
+    unique_opp_teams = opposing_teams.map do |set|
+      set[1]
+    end.uniq
+    unique_opp_teams.map do |opp_team_id|
+      papa_bear[opp_team_id] ||= []
+      papa_bear[opp_team_id] = season_win_data_set.count do |set|
+        set[1] == opp_team_id
+      end
+    end
+    unique_opp_teams.map do |opp_team_id|
+      mama_bear[opp_team_id] ||= []
+      mama_bear[opp_team_id] = opposing_teams.count do |set|
+        set[1] == opp_team_id
+      end
+    end
+    mama_bear.map do |opp_team_id, wins|
+      mama_bear[opp_team_id] = (wins.to_f/papa_bear[opp_team_id])
+    end
+    teeny_bear = mama_bear.max_by {|opp_team_id, win_percentage| win_percentage}[0]
+    team_id_to_team_name(teeny_bear)
+  end
+
+  def rival(team_id)
+    papa_bear = {}
```

Comment:
Great work on this, we will need to refactor these together, but that is what is next. Mama bear, Pape bear, and baby bear forever! Oh and teeny bear.